### PR TITLE
Bump Hub terraform to 0.12.7

### PR DESCRIPTION
**Pull request by brenetic**
🤞

### CODE REVIEW

File: terraform/modules/hub/hub_frontend.tf [hunk 1]
```diff
@@ -48,98 +48,98 @@ locals {
   }
```
Comment (philandstuff) [inline terraform/modules/hub/hub_frontend.tf]: Note: terraform 0.12 correctly dedents this heredoc, where terraform 0.11 didn't.  This causes a rebuild of the frontend task definition.  This is ok though and is generally a change for the better.

File: terraform/modules/hub/analytics.tf [hunk 2]
```diff
@@ -24,31 +24,31 @@ locals {
 
   LOCATIONS
 
-  nginx_analytics_location_blocks_base64 = "${base64encode(local.analytics_location_blocks)}"
+  nginx_analytics_location_blocks_base64 = base64encode(local.analytics_location_blocks)
 }
 
 data "template_file" "analytics_task_def" {
   template = "${file("${path.module}/files/tasks/analytics.json")}"
```
Comment (philandstuff) [inline terraform/modules/hub/analytics.tf]:
```suggestion
  template = file("${path.module}/files/tasks/analytics.json")
```

File: terraform/modules/hub/beat_exporter.tf [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-  deployment       = "${var.deployment}"
+  deployment       = var.deployment
   service_name     = "beat-exporter"
   image_name       = "verify-beat-exporter"
-  tools_account_id = "${var.tools_account_id}"
+  tools_account_id = var.tools_account_id
 }
 
 data "template_file" "beat_exporter_task_def" {
   template = "${file("${path.module}/files/tasks/beat-exporter.json")}"
```
Comment (philandstuff) [inline terraform/modules/hub/beat_exporter.tf]:
```suggestion
   template = file("${path.module}/files/tasks/beat-exporter.json")
```

File: terraform/modules/hub/cloudwatch_exporter.tf [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
+  tools_account_id = var.tools_account_id
 }
 
 data "template_file" "cloudwatch_exporter_task_def" {
   template = "${file("${path.module}/files/tasks/cloudwatch-exporter.json")}"
 
-  vars {
+  vars = {
     image_identifier = "${local.tools_account_ecr_url_prefix}-verify-cloudwatch-exporter@${var.cloudwatch_exporter_image_digest}"
     config_base64    = "${base64encode(file("${path.module}/files/prometheus/cloudwatch_exporter.yml"))}"
```
Comment (philandstuff) [inline terraform/modules/hub/cloudwatch_exporter.tf]:
```suggestion
    config_base64    = base64encode(file("${path.module}/files/prometheus/cloudwatch_exporter.yml"))
```

File: terraform/modules/hub/egress_proxy.tf [hunk 5]
```diff
@@ -45,42 +45,42 @@ resource "aws_security_group_rule" "egress_proxy_instance_egress_to_internet_ove
...
+    replace(local.event_emitter_api_gateway[0], ".", "\\."), # API Gateway
+    var.splunk_hostname,                                     # Splunk
   ]
 
-  egress_proxy_whitelist = "${join(" ", local.egress_proxy_whitelist_list)}"
+  egress_proxy_whitelist = join(" ", local.egress_proxy_whitelist_list)
 }
 
 data "template_file" "egress_proxy_task_def" {
   template = "${file("${path.module}/files/tasks/squid.json")}"
```
Comment (philandstuff) [inline terraform/modules/hub/egress_proxy.tf]:
```suggestion
  template = file("${path.module}/files/tasks/squid.json")
```

File: terraform/modules/hub/hub_frontend.tf [hunk 6]
```diff
@@ -48,98 +48,98 @@ locals {
...
   LOCATIONS
 
-  location_blocks_base64 = "${base64encode(local.location_blocks)}"
+  location_blocks_base64 = base64encode(local.location_blocks)
 }
 
 data "template_file" "frontend_task_def" {
   template = "${file("${path.module}/files/tasks/frontend.json")}"
```
Comment (philandstuff) [inline terraform/modules/hub/hub_frontend.tf]:
```suggestion
  template = file("${path.module}/files/tasks/frontend.json")
```

File: terraform/modules/hub/hub_metadata.tf [hunk 7]
```diff
@@ -12,48 +12,48 @@ resource "aws_security_group" "metadata_task" {
 data "template_file" "metadata_task_def" {
   template = "${file("${path.module}/files/tasks/metadata.json")}"
```
Comment (philandstuff) [inline terraform/modules/hub/hub_metadata.tf]:
```suggestion
  template = file("${path.module}/files/tasks/metadata.json")
```

File: terraform/modules/hub/modules/ecs_asg/asg.tf [hunk 8]
```diff
@@ -17,28 +17,28 @@ locals {
 data "template_file" "cloud_init" {
   template = "${file("${path.module}/files/cloud-init.sh")}"
```
Comment (philandstuff) [inline terraform/modules/hub/modules/ecs_asg/asg.tf]:
```suggestion
  template = file("${path.module}/files/cloud-init.sh")
```

**Comment by brenetic** [comment]
thanks @philandstuff have updated

**Pull request merged**